File: leetcode/cpp/747_Largest_Number_At_Least_Twice_of_Others.cpp
#include <vector>
using namespace std;
class Solution {
public:
	int dominantIndex(vector<int>& nums) {
		int max_val = INT_MIN, max_idx = -1;
		int second_max = INT_MIN;

		for (int i = 0; i < nums.size(); ++i) {
			int n = nums[i];
			if (n > max_val) {
				second_max = max_val;
				max_val = n;
				max_idx = i;
			} else if (n > second_max) {
				second_max = n;
			}

		}

		return max_val >= second_max << 1 ? max_idx : -1;
	}
};

File: leetcode/cpp/747_Largest_Number_At_Least_Twice_of_Others.py
class Solution:
    def dominantIndex(self, nums: List[int]) -> int:
        max_val = float('-inf')
        sec_max = float('-inf')
        max_idx = -1

        for i, n in enumerate(nums):
            if n > max_val:
                sec_max = max_val
                max_val = n
                max_idx = i
            elif n > sec_max:
                sec_max = n

        return max_idx if max_val >= 2 * sec_max else -1

